src/app/(main)/my-voyage/[teamId]/directory/components/DirectoryComponentWrapper.tsx

```diff
@@ -0,0 +1,135 @@
+import { redirect } from "next/navigation";
+import DirectoryProvider from "./DirectoryProvider";
+import TeamMember from "./TeamMember";
+import { TeamDirectory } from "@/store/features/directory/directorySlice";
+import Banner from "@/components/banner/Banner";
+import { getAccessToken } from "@/utils/getCookie";
+import { AsyncActionResponse, handleAsync } from "@/utils/handleAsync";
+import { GET } from "@/utils/requests";
+import { CacheTag } from "@/utils/cacheTag";
+import { User, VoyageTeamMember } from "@/store/features/user/userSlice";
+import { getUser } from "@/utils/getUser";
+import { getTimezone } from "@/utils/getTimezone";
+import VoyagePageBannerContainer from "@/components/banner/VoyagePageBannerContainer";
+
+interface FetchTeamDirectoryProps {
+  teamId: number;
+  user: User | null;
+}
+
+export async function fetchTeamDirectory({
+  teamId,
+  user,
+}: FetchTeamDirectoryProps): Promise<AsyncActionResponse<TeamDirectory>> {
+  const token = getAccessToken();
+
+  const fetchTeamDirectoryAsync = () =>
+    GET<TeamDirectory>(
+      `api/v1/teams/${teamId}`,
+      token,
+      "force-cache",
+      CacheTag.directory,
+    );
+
+  const [res, error] = await handleAsync(fetchTeamDirectoryAsync);
+
+  if (res) {
+    updateDirectoryWithCurrentTime(res);
+    const teamMember = res.voyageTeamMembers;
+    const elementToSort = teamMember.find(
+      (element) => element.member.discordId === user?.discordId,
+    );
+    moveElementToFirst(teamMember, elementToSort);
+  }
+
+  return [res, error];
+}
+
+function updateDirectoryWithCurrentTime(data: TeamDirectory) {
+  return data.voyageTeamMembers.forEach((teamMember) => {
+    const { timezone } = teamMember.member;
+    const currentTime = getTimezone(timezone);
+    teamMember.member.currentTime = currentTime;
+  });
+}
+
+function moveElementToFirst<T>(arr: T[], element: T): T[] {
+  const index = arr.indexOf(element);
+  if (index === -1) {
+    return arr;
+  }
+  [arr[index], arr[0]] = [arr[0], arr[index]];
+  return arr;
+}
+
+interface TeamDirectoryProps {
+  params: {
+    teamId: string;
+  };
+}
+
+export default async function DirectoryComponentWrapper({
+  params,
+}: TeamDirectoryProps) {
+  let teamDirectory: TeamDirectory;
+  let currentVoyageTeam: VoyageTeamMember | undefined;
+
+  const [user, error] = await getUser();
+
+  if (user) {
+    currentVoyageTeam = user.voyageTeamMembers.find(
+      (voyage) => voyage.voyageTeam.voyage.status.name === "Active",
+    );
+  }
+
+  if (error) {
+    return `Error: ${error?.message}`;
```

Comment:
https://nextjs.org/docs/app/building-your-application/routing/error-handling#handling-server-errors

Nextjs strips error messages in production and gives a very bland gibberish message instead. This is not helpful whatsoever because we need to display errors and with the way the project is structured (and what I'm assuming is the way to do it with next server components), the way the errors are handled is as outlined in the docs I linked. Only way I found to handle this was to return the error instead of throwing it. Thrown errors are swallowed.

Imagine this, the user tries to login and there's some kind of error from the server and they get this message (this is what gets output): "An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error." That's not helpful at all.